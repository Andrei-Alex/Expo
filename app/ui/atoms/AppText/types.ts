import { TextStyle } from 'react-native';

export type Props = {
  children: React.ReactNode;
  style?: TextStyle;
};
