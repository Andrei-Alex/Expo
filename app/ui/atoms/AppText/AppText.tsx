import React from "react";
import {Text} from "react-native";

import {styles, Props} from ".";

const AppText: React.FC<Props> = ({children}) => {
    return (
        <Text style={styles.text}>
            {children}
        </Text>
    );
}

export default AppText;
