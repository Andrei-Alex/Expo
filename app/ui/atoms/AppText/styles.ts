import { StyleSheet, Platform } from "react-native";
import {isAndroid} from "../../../config";
export const styles = StyleSheet.create({
text: {
    fontSize: 18,
    fontFamily: isAndroid ? 'Roboto' : 'Avenir',
}
});
