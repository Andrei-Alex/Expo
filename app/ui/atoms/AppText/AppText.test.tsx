import * as React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';

import AppText from './AppText';

describe('AppText', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<AppText>Test</AppText>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('applies style correctly', () => {
    const testStyle = { fontSize: 20, color: 'blue' };
    const tree = renderer
      .create(<AppText style={testStyle}>Styled Test</AppText>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('applies passed TextStyle', () => {
    const testStyle = { color: 'red', fontSize: 20 };

    const instance = renderer.create(
      <AppText style={testStyle}>Test</AppText>
    ).root;
    const textComponent = instance.findByType(Text);

    const receivedStyle = textComponent.props.style;
    const styleArray = Array.isArray(receivedStyle)
      ? receivedStyle
      : [receivedStyle];

    const hasTestStyle = styleArray.some((style) =>
      Object.entries(testStyle).every(([key, value]) => style[key] === value)
    );
    expect(hasTestStyle).toBeTruthy();
  });
});
