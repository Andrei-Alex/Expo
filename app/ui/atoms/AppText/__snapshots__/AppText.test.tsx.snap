// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppText applies style correctly 1`] = `
<Text
  style={
    [
      {
        "fontFamily": "Avenir",
        "fontSize": 20,
      },
      {
        "color": "blue",
        "fontSize": 20,
      },
    ]
  }
>
  Styled Test
</Text>
`;

exports[`AppText renders correctly 1`] = `
<Text
  style={
    [
      {
        "fontFamily": "Avenir",
        "fontSize": 20,
      },
      undefined,
    ]
  }
>
  Test
</Text>
`;
